app/src/main/java/com/github/se/travelpouch/ui/notifications/NotificationsScreen.kt

```diff
@@ -0,0 +1,180 @@
+                        },
+                        onFailure = { e ->
+                          Toast.makeText(context, "Error: ${e.message}", Toast.LENGTH_SHORT).show()
+                        })
+                  },
+                  modifier = Modifier.testTag("DeleteAllNotificationsButton"),
+              ) {
+                Text("Delete All")
+              }
+            },
+        )
+      },
+      content = { paddingValues ->
+        LazyColumn(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(paddingValues)
+                    .testTag("LazyColumnNotificationsScreen")) {
+              categorizedNotifications.forEach { (category, notifications) ->
+                item {
+                  Text(
+                      text = category,
+                      fontSize = 20.sp,
+                      fontWeight = FontWeight.Bold,
+                      modifier = Modifier.padding(8.dp).testTag("CategoryTitle"))
+                }
+                itemsIndexed(notifications) { _, notification ->
+                  notificationViewModel.markNotificationAsRead(notification.notificationUid)
+                  NotificationItem(
+                      notification = notification,
+                      notificationViewModel = notificationViewModel,
+                      profileViewModel = profileModelView,
+                      listTravelViewModel = listTravelViewModel,
+                      navigationActions = navigationActions,
+                      activityViewModel = activityViewModel,
+                      documentViewModel = documentViewModel,
+                      eventsViewModel = eventsViewModel)
+                }
+              }
+            }
+      },
+      bottomBar = {
+        BottomNavigationMenu(
+            navigationActions = navigationActions,
+            tabList =
+                listOf(
+                    TopLevelDestinations.NOTIFICATION,
+                    TopLevelDestinations.TRAVELS,
+                    TopLevelDestinations.CALENDAR))
+      })
+}
+
+fun categorizeNotifications(notifications: List<Notification>): Map<String, List<Notification>> {
+  val thisWeek = mutableListOf<Notification>()
+  val lastWeek = mutableListOf<Notification>()
+  val lastMonth = mutableListOf<Notification>()
+  val lastYear = mutableListOf<Notification>()
+
+  val now = Calendar.getInstance()
+
+  notifications.forEach { notification ->
+    val notificationTime = notification.timestamp.toDate()
+
+    when {
+      isThisWeek(notificationTime, now) -> thisWeek.add(notification)
+      isLastWeek(notificationTime, now) -> lastWeek.add(notification)
+      isLastMonth(notificationTime, now) -> lastMonth.add(notification)
+      isLastYear(notificationTime, now) -> lastYear.add(notification)
+    }
+  }
+
+  return mapOf(
+      "This week" to thisWeek,
+      "Last week" to lastWeek,
+      "Last month" to lastMonth,
+      "Last year" to lastYear)
+}
```

Comment:
it may be better to add a category "older". But this optional we can do it later